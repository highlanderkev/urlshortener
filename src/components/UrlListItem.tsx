import React from "react";
import { URLEntry } from "../urlEntry.model";

type UrlListItemProps = {
  item: URLEntry;
  onRemove: Function;
};

export default class UrlListItem extends React.Component<UrlListItemProps> {
  handleClick = () => {
    this.props.onRemove(this.props.item.slug);
  };

  render() {
    return (
      <li
        key={this.props.item.slug}
        className="list-group-item d-flex justify-content-between align-items-start"
      >
        <div className="ms-2 me-auto">
          <div className="fw-bold">
            <a href={this.props.item.url}>{this.props.item.url}</a>
            <br />
            <a href={this.props.item.short_url}>{this.props.item.short_url}</a>
          </div>
          <small>{this.props.item.slug}</small>
        </div>
        <button
          type="button"
          className="btn-close"
          aria-label="Close"
          onClick={this.handleClick}
        ></button>
      </li>
    );
  }
}
