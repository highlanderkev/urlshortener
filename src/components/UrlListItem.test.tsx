import React from 'react';
import UrlListItem from "./UrlListItem";
import renderer from "react-test-renderer";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()})
describe('URL List Item Component', () => {
  it("matches snapshot", () => {
    const listItem = renderer.create(<UrlListItem key={'test'} item={{url: 'test', short_url: 'test', slug: 'test'}} onRemove={() => {}} />);
    let tree = listItem.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("displays list url entry", () => {
    const listItem = shallow(<UrlListItem key={'test'} item={{url: 'test', short_url: 'test', slug: 'test'}} onRemove={() => {}} />)
    expect(listItem.text()).toEqual("testtesttest");
  });

})
