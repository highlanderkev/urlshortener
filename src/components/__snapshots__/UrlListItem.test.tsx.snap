// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`URL List Item Component matches snapshot 1`] = `
<li
  className="list-group-item d-flex justify-content-between align-items-start"
>
  <div
    className="ms-2 me-auto"
  >
    <div
      className="fw-bold"
    >
      <a
        href="test"
      >
        test
      </a>
      <br />
      <a
        href="test"
      >
        test
      </a>
    </div>
    <small>
      test
    </small>
  </div>
  <button
    aria-label="Close"
    className="btn-close"
    onClick={[Function]}
    type="button"
  />
</li>
`;
